crates/client/rpc/src/errors.rs

```diff
@@ -62,3 +62,15 @@ impl From<StarknetRpcApiError> for jsonrpsee::core::Error {
         jsonrpsee::core::Error::Call(CallError::Custom(ErrorObject::owned(err as i32, err.to_string(), None::<()>)))
     }
 }
+
+pub fn rpc_error_with_data<T: sp_runtime::Serialize>(
+    err: StarknetRpcApiError,
+    data: T,
+    msg: String,
+) -> jsonrpsee::core::Error {
+    jsonrpsee::core::Error::Call(CallError::Custom(ErrorObject::owned(
+        err as i32,
+        format!("{err}{msg}"),
```

Comment:
Thanks for the review!

In the latest commit, it is now possible to distinguish between `revert_error` and other errors within `SimulationError`. Therefore, I believe it may not be necessary to request this PR based on the current content, even if I modify it to reflect your review.

One question though: is there somewhere I can find the spec for the revert condition?

Reply:
What do you mean? Sorry I don't get your point

Reply:
I apologize if my previous comment was unclear. Since PR #1555  implements similar error handling, I suggest we close this PR. And if there are any further requirements, I would be happy to address them in the future in a new PR